settings.gradle

```diff
@@ -0,0 +1,19 @@
+rootProject.name = 'kiota-java'
+
+include ':components:abstractions'
```

Comment:
out of curiosity why not put this block in the else instead of logging a warning?

Reply:
if I understood correctly the current setup those modules are always available, but the android modules should be compiled(in addition to the standard) only if the relevant SDK is installed.

Putting those in the `else` branch disables the modules from being built/tested/compiled when you have the SDK installed, is it the expected behavior?

Just to make it clear, I have almost 0 experience with Android development, and I'm refactoring with plain Java usage in mind, feel free to challenge and eventually correct my decisions and assumptions if you feel like 👍

Reply:
right, sorry I was multi-tasking, and I'm not great at doing so.
the idea of the android projects is just to check we're not using APIs from API levels above the one we're targeting (android only has a partial implementation of the java 8 spec and they release additional implementations with newer API levels).
during our CI we run both (with different workflows), but during publish only the "pure java" project is being used.

With that context I think we can simply remove the reference to the android projects here, and add a new root project in the android directory, makes sense?

Reply:
@baywet thanks a lot for this explanation, very much needed!
I think that this commit correctly serves the purpose: https://github.com/microsoft/kiota-java/pull/118/commits/3e2e9962d573a1259487cb5c6d2f9fbff0992247

We can probably simplify the android setup, but let's keep it for another PR.